sdk/mobile-center-distribute/src/main/java/com/microsoft/azure/mobile/distribute/Distribute.java

```diff
@@ -839,41 +855,39 @@ private synchronized void showUpdateDialog() {
         AlertDialog.Builder dialogBuilder = new AlertDialog.Builder(mForegroundActivity);
         dialogBuilder.setTitle(R.string.mobile_center_distribute_update_dialog_title);
         final ReleaseDetails releaseDetails = mReleaseDetails;
-        String releaseNotes = releaseDetails.getReleaseNotes();
-        if (TextUtils.isEmpty(releaseNotes))
-            dialogBuilder.setMessage(R.string.mobile_center_distribute_update_dialog_message);
-        else
-            dialogBuilder.setMessage(releaseNotes);
+        String message;
+        if (releaseDetails.isMandatoryUpdate()) {
+            message = mContext.getString(R.string.mobile_center_distribute_update_dialog_message_mandatory);
+        } else {
+            message = mContext.getString(R.string.mobile_center_distribute_update_dialog_message_optional);
+        }
+        message = String.format(message, getAppName(), mReleaseDetails.getShortVersion(), mReleaseDetails.getVersion());
+        dialogBuilder.setMessage(message);
         dialogBuilder.setPositiveButton(R.string.mobile_center_distribute_update_dialog_download, new DialogInterface.OnClickListener() {
 
             @Override
             public void onClick(DialogInterface dialog, int which) {
                 enqueueDownloadOrShowUnknownSourcesDialog(releaseDetails);
             }
         });
-        if (releaseDetails.isMandatoryUpdate()) {
-            dialogBuilder.setCancelable(false);
-        } else {
-            dialogBuilder.setNegativeButton(R.string.mobile_center_distribute_update_dialog_ignore, new DialogInterface.OnClickListener() {
+        dialogBuilder.setCancelable(false);
+        if (!releaseDetails.isMandatoryUpdate()) {
+            dialogBuilder.setNegativeButton(R.string.mobile_center_distribute_update_dialog_postpone, new DialogInterface.OnClickListener() {
 
                 @Override
                 public void onClick(DialogInterface dialog, int which) {
-                    ignoreRelease(releaseDetails);
+                    postponeRelease(releaseDetails);
                 }
             });
-            dialogBuilder.setNeutralButton(R.string.mobile_center_distribute_update_dialog_postpone, new DialogInterface.OnClickListener() {
-
-                @Override
-                public void onClick(DialogInterface dialog, int which) {
-                    completeWorkflow(releaseDetails);
-                }
-            });
-            setOnCancelListener(dialogBuilder, releaseDetails);
         }
         mUpdateDialog = dialogBuilder.create();
         showAndRememberDialogActivity(mUpdateDialog);
     }
 
+    private String getAppName() {
```

Comment:
So using app name from bundle as better for localization. Designers will talk to distribute team to improve portal first time install experience to match app.